sdk/python/kfp/dsl/_container_op.py

```diff
@@ -1371,6 +1375,25 @@ def add_node_selector_constraint(self, label_name: str,
     super(ContainerOp, self).add_node_selector_constraint(label_name, value)
     return self
 
+  def add_cpu_request(self, cpu: str) -> 'ContainerOp':
+    """Adds a cpu request to which can be a parameter
+
+    Args:
+        cpu (str): kubernetes cpu request, https://kubernetes.io/docs/tasks/configure-pod-container/assign-cpu-resource/#specify-a-cpu-request-and-a-cpu-limit
+    """
+
+    self.cpu_request = cpu
```

Comment:
Do we want to do any kind of validation? so we can capture user input error at compile time rather than runtime?

Reply:
One important requirement was that the resource request can be a pipeline parameter, so compile time check is not possible

Reply:
I see. The way this is written right now implies that it doesn't accept pipeline parameter. 
To accept both constant value and pipeline parameter. The type annotation should be `Union[str, PipelineParam]` instead of just `str`.  And the docstring should mention the same.
And we still can (and probably should) do validation when possible. See below for example:
https://github.com/kubeflow/pipelines/blob/96325090490ff2d8f8a3a7b5036f9a7056a46e69/sdk/python/kfp/dsl/_container_op.py#L229-L246